### PR TITLE
[RELEASE] SRF v22.11

### PR DESCRIPTION
**Pull request by mdemoret-nv**
### What does this mean?
Only critical/hotfix level issues should be merged into `branch-22.11` until release (merging of this PR).

### What is the purpose of this PR?
- Update documentation
- Allow testing for the new release
- Enable a means to merge `branch-22.11` into main for the release

### CODE REVIEW

**Review by cwharris — APPROVED** [review]

File: python/srf/tests/sample_modules.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        "TemplateModuleInt", "srf_unittest", PybindSegmentModuleVersion);
+    modules::ModelRegistryUtil::create_registered_module<srf::modules::TemplateModule<std::string>>(
+        "TemplateModuleString", "srf_unittest", PybindSegmentModuleVersion);
+
+    std::stringstream sstream;
+    sstream << srf_VERSION_MAJOR << "." << srf_VERSION_MINOR << "." << srf_VERSION_PATCH;
+
+    module.attr("__version__") = sstream.str();
+}
+}  // namespace srf::pysrf
```
Comment (drobison00) [inline python/srf/tests/sample_modules.cpp:78]:
```suggestion
}  // namespace srf::pysrf

```

File: python/srf/tests/sample_modules.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        "NestedModule", "srf_unittest", PybindSegmentModuleVersion);
+    modules::ModelRegistryUtil::create_registered_module<srf::modules::TemplateModule<int>>(
+        "TemplateModuleInt", "srf_unittest", PybindSegmentModuleVersion);
+    modules::ModelRegistryUtil::create_registered_module<srf::modules::TemplateModule<std::string>>(
+        "TemplateModuleString", "srf_unittest", PybindSegmentModuleVersion);
+
+    std::stringstream sstream;
+    sstream << srf_VERSION_MAJOR << "." << srf_VERSION_MINOR << "." << srf_VERSION_PATCH;
+
+    module.attr("__version__") = sstream.str();
```
Comment (drobison00) [inline python/srf/tests/sample_modules.cpp:76]:
```suggestion
    module.attr("__version__") =
        SRF_CONCAT_STR(srf_VERSION_MAJOR << "." << srf_VERSION_MINOR << "." << srf_VERSION_PATCH);
```

Comment (drobison00) [reply python/srf/tests/sample_modules.cpp:76]: Missed switching to SRF_CONCAT_STR in two files.

File: python/srf/tests/test_edges.cpp [hunk 3]
```diff
@@ -266,10 +268,9 @@ PYBIND11_MODULE(test_edges_cpp, m)
...
 
-#ifdef VERSION_INFO
-    m.attr("__version__") = MACRO_STRINGIFY(VERSION_INFO);
-#else
-    m.attr("__version__") = "dev";
-#endif
+    std::stringstream sstream;
+    sstream << srf_VERSION_MAJOR << "." << srf_VERSION_MINOR << "." << srf_VERSION_PATCH;
+
+    module.attr("__version__") = sstream.str();
```
Comment (drobison00) [inline python/srf/tests/test_edges.cpp:274]:
```suggestion
    module.attr("__version__") =
        SRF_CONCAT_STR(srf_VERSION_MAJOR << "." << srf_VERSION_MINOR << "." << srf_VERSION_PATCH);
```

File: python/srf/tests/utils.cpp [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+            throw std::runtime_error(msg);
+        },
+        py::arg("msg") = "");
+
+#ifdef VERSION_INFO
+    m.attr("__version__") = MACRO_STRINGIFY(VERSION_INFO);
+#else
+    module.attr("__version__") = "dev";
+#endif
```
Comment (drobison00) [inline python/srf/tests/utils.cpp:51]:
```suggestion
    module.attr("__version__") =
        SRF_CONCAT_STR(srf_VERSION_MAJOR << "." << srf_VERSION_MINOR << "." << srf_VERSION_PATCH);
```

**Review by drobison00 — APPROVED** [review]
Added a couple last minute tweaks if we want to get them in.

**Pull request merged**